pubmed_dl/ncbi.py

```diff
@@ -112,28 +113,31 @@ def uids_to_docs(uids: List[str]) -> List[Dict[str, Collection[Any]]]:
         upper = min([lower + MAX_EFETCH_RETMAX, num_uids])
         id = uids[lower:upper]
         try:
+            if "bruh" in id:
+                raise e
             eutil_response = _get_eutil_records("efetch", id, rettype="medline", retmode="text")
         except Exception as e:
             print(f"Error encountered in uids_to_docs {e}")
             raise e
         else:
             output = _medline_to_docs(eutil_response)
             docs = docs + output
-    return docs
+    yield docs

```

Comment:
I think you want something more like:

```
for i in range(num_queries):
        lower = i * MAX_EFETCH_RETMAX
        ...
        try:
            ...
        else:
            yield output
```

Reply:
Yup the yield should be inside the for loop

Reply:
Perfect!! This works.

Reply:
@Anwesh1 Make sure to actually commit the change